lib/internal/errors.js

```diff
@@ -0,0 +1,282 @@
+
+// Utility methods for the messages above
+
+function bufferTooBig(max) {
+  const assert = lazyAssert();
+  assert(max !== undefined);
+  return 'Cannot create final Buffer. It would be larger ' +
+         `than 0x${max.toString(16)} bytes`;
+}
+
+function fileTooBig(arg) {
+  const assert = lazyAssert();
+  assert(arg !== undefined);
+  return 'File size is greater than possible Buffer: ' +
+         `0x${arg.toString(16)} bytes`;
+}
+
+function unsupportedProtocol(protocol, expected) {
+  let msg = `Protocol "${protocol}" is not supported`;
+  if (expected) {
+    msg += `. Expected "${expected}"`;
+  }
+  return msg;
+}
+
+function notImplemented(additional) {
+  let msg = 'Not implemented';
+  if (additional)
+    msg += `: ${additional}`;
+  return msg;
+}
+
+function invalidOptionValue(opt, val) {
+  const assert = lazyAssert();
+  assert(opt);
+  let msg = `Invalid value for "${opt}" option`;
+  if (val)
+    msg += `: ${val}`;
+  return msg;
+}
+
+function invalidArgumentValue(arg, val) {
+  const assert = lazyAssert();
+  assert(arg);
+  let msg = `Invalid value for "${arg}" argument`;
+  if (val)
+    msg += `: ${val}`;
+  return msg;
+}
+
+function argNotNum(arg) {
+  const assert = lazyAssert();
+  assert(arg);
+  return `"${arg}" argument must not be a number`;
+}
+
+function requiredArgument(arg) {
+  const assert = lazyAssert();
+  assert(arg);
+  return `"${arg}" argument is required and cannot be undefined`;
+}
+
+function invalidArgument(arg, expected) {
+  return invalidOptionOrArgument(true, arg, expected);
+}
+
+function invalidOption(opt, expected) {
+  return invalidOptionOrArgument(false, opt, expected);
+}
+
+function formatArg(arg) {
+  if (typeof arg === 'number') {
+    arg >>>= 0;
+    switch (arg) {
+      case 1:
+        arg = 'The first';
+        break;
+      case 2:
+        arg = 'The second';
+        break;
+      case 3:
+        arg = 'The third';
+        break;
+      default:
+        arg = `The ${arg}th`;
+    }
+  } else {
+    arg = `"${arg}"`;
+  }
+  return arg;
+}
+
+// This takes an arg name and one or more expected types to generate an
+// appropriate error message indicating that the given argument must be
+// of the expected type. For instance:
+//  "foo" argument must be a string
+//  "bar" argument must be an object
+//  "baz" argument must be one of: string, object
+//  first argument must be a string
+//  ... and so forth.
+function invalidOptionOrArgument(isargument, arg, expected) {
+  const assert = lazyAssert();
+  const util = lazyUtil();
+  assert(arg && expected);
+  // Is this error message for an argument or an option?
+  const what = isargument ? 'argument' : 'option';
+  if (typeof expected === 'string') {
+    // Check to see if we need "a" or "an".
+    const ch = expected.codePointAt(0) | 0x20;
+    const prefix = (ch === 0x61 || ch === 0x65 ||
+                    ch === 0x69 || ch === 0x6f ||
+                    ch === 0x75) ? 'an' : 'a';
```

Comment:
I think this returns false positives for e.g. `Uint8Array` – maybe the message should just read something like `must be of type ${expected}`?